src/Oscoin/API/HTTP/Client.hs

```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE UndecidableInstances #-}
```

Comment:
Would you say this can be avoided with a bit more work - ie. is this debt, or a necessary evil?

Reply:
This is a necessary evil, because we need to expand the type family `Hash` in the LHS of our constraints. I was for a long time scared by `UndecidableInstances`, but it's actually a benign extension in my experience. Cfr. https://monadfix.io/#2017-12-18-undecidableinstances